src/services/search/listingSearch.ts

```diff
@@ -128,11 +191,88 @@ export const fetchListings = async ({
     path: "listings/search",
     query,
     options,
+    defaultSort: defaultUserSort,
+    defaultPagination: defaultUserPagination,
+  })
+
+  return sanitizeListingResponse(response)
+}
+
+export const fetchDealerListings = async ({
+  dealerId,
+  query = {},
+  options = {},
+}: {
+  dealerId: number
+  query?: ListingQueryParams
+  options?: ApiCallOptions
+}): Promise<Paginated<SearchListing>> => {
+  const response = await searchForListings({
+    path: `dealers/${dealerId}/listings/search`,
+    query,
+    options: {
+      isAuthorizedRequest: true,
+      ...options,
+    },
+    defaultSort: defaultDealerSort,
+    defaultPagination: defaultDealerPagination,
   })
 
   return sanitizeListingResponse(response)
 }
 
+const toCamelCase = (snakeCase) => {
+  const [w, ...ws] = snakeCase.split("_").map((word) => {
+    const [l, ...ls] = word.toLowerCase()
+
+    return [l.toUpperCase(), ...ls].join("")
+  })
+
+  return [w.toLowerCase(), ...ws].join("")
+}
+
+export const fetchDealerArchivedListings = async ({
+  dealerId,
+  query = {},
+  options = {},
+}: {
+  dealerId: number
+  query?: ListingSortParams & PaginationParams
+  options?: ApiCallOptions
+}): Promise<Paginated<SearchListing>> => {
+  const { page, size, sortOrder, sortType } = query
+
+  const sortOrDefault = {
+    sortType: sortType || defaultDealerSort.sortType,
+    sortOrder: sortOrder || defaultDealerSort.sortOrder,
+  }
+
+  const queryParams = {
+    page: pageOrDefault(page, defaultDealerPagination),
+    size: sizeOrDefault(size, defaultDealerPagination),
+    sort: `${toCamelCase(sortOrDefault.sortType)},${toCamelCase(
+      sortOrDefault.sortOrder
+    )}`,
+  }
+
+  const { content, ...response } = await fetchPath({
+    path: `dealers/${dealerId}/archived-listings${
```

Comment:
You could also generate the query string and leave the key handling to the `toQueryString` as you should get an empty string if there are no keys.... And only check if we have a qs here

Reply:
Oh the check itself doesn't even make sense since there always are parameters (keys). Simplified.